src/sentry/api/serializers/models/discoversavedquery.py

```diff
@@ -38,6 +45,13 @@ def serialize(self, obj, attrs, user, **kwargs):
             if obj.query.get(key) is not None:
                 data[key] = obj.query[key]
 
+        # expire queries that are beyond the retention period
```

Comment:
Replicated [this](https://github.com/getsentry/sentry/blob/master/src/sentry/utils/snuba.py#L508) logic

Reply:
Instead of replicating, could we move this logic into a util function? Might be a good fit in `sentry.utils.dates`

Reply:
Nice, I like that this updates the start for you too now so that the query stays within retention when possible